frame/conviction-voting/src/lib.rs

```diff
@@ -0,0 +1,625 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Voting Pallet
+//!
+//! - [`Config`]
+//! - [`Call`]
+//!
+//! ## Overview
+//!
+//! Pallet for managing actual voting in polls.
+
+#![recursion_limit = "256"]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+	ensure,
+	traits::{
+		fungible, Currency, Get, LockIdentifier, LockableCurrency, PollStatus, Polling,
+		ReservableCurrency, WithdrawReasons,
+	},
+};
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, Saturating, Zero},
+	ArithmeticError, DispatchError, DispatchResult, Perbill,
+};
+use sp_std::prelude::*;
+
+mod conviction;
+mod types;
+mod vote;
+pub mod weights;
+pub use conviction::Conviction;
+pub use pallet::*;
+pub use types::{Delegations, Tally, UnvoteScope};
+pub use vote::{AccountVote, Casting, Delegating, Vote, Voting};
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+const CONVICTION_VOTING_ID: LockIdentifier = *b"pyconvot";
+
+type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+type VotingOf<T> = Voting<
+	BalanceOf<T>,
+	<T as frame_system::Config>::AccountId,
+	<T as frame_system::Config>::BlockNumber,
+	PollIndexOf<T>,
+>;
+#[allow(dead_code)]
+type DelegatingOf<T> = Delegating<
+	BalanceOf<T>,
+	<T as frame_system::Config>::AccountId,
+	<T as frame_system::Config>::BlockNumber,
+>;
+pub type TallyOf<T> = Tally<BalanceOf<T>, <T as Config>::MaxTurnout>;
+pub type VotesOf<T> = BalanceOf<T>;
+type PollIndexOf<T> = <<T as Config>::Polls as Polling<TallyOf<T>>>::Index;
+#[cfg(feature = "runtime-benchmarks")]
+type IndexOf<T> = <<T as Config>::Polls as Polling<TallyOf<T>>>::Index;
+type ClassOf<T> = <<T as Config>::Polls as Polling<TallyOf<T>>>::Class;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::DispatchResult;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
```

Comment:
```suggestion
	#[pallet::generate_store(pub(super) trait Store)]
	#[pallet::generate_storage_info]
	pub struct Pallet<T>(_);
```

This will be included by default soon. See https://github.com/paritytech/substrate/pull/10662

Reply:
This will lead to some errors probably, I can take care of them, just ping me if you want me to do it.